File: 69.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x == 0:
            return 0
        i = 1
        while(i <= x):
            if i * i > x:
                return i-1
            i += 1


s = Solution()
print s.mySqrt(2)
print s.mySqrt(4)
print s.mySqrt(8)

File: 69.go
package main

import "fmt"

func main() {
	fmt.Println(mySqrt(8))
	fmt.Println(mySqrt(6))
	fmt.Println(mySqrt(4))
	fmt.Println(mySqrt(3))
	fmt.Println(mySqrt(2))
	fmt.Println(mySqrt(0))
}

func mySqrt(x int) int {
	l, r := 0, x
	ans := -1
	for l <= r {
		mid := (r-l)/2 + l
		if mid*mid <= x {
			ans = mid
			l = mid + 1
		} else {
			r = mid - 1
		}
	}
	return ans
}
